src/controllers/mainController.ts

```diff
@@ -865,9 +847,56 @@ export default class MainController implements vscode.Disposable {
      * Called by VS Code when user settings are changed
      * @param ConfigurationChangeEvent event that is fired when config is changed
      */
-    public onDidChangeConfiguration(e: vscode.ConfigurationChangeEvent): void {
+    public async onDidChangeConfiguration(e: vscode.ConfigurationChangeEvent): Promise<void> {
         if (e.affectsConfiguration(Constants.extensionName)) {
-            this._objectExplorerProvider.refresh(undefined);
+            let needsRefresh = false;
+            // user connections is a super set of object explorer connections
+            let userConnections: any[] = this._vscodeWrapper.getConfiguration(Constants.extensionName).get(Constants.connectionsArrayName);
+            let objectExplorerConnections = this._objectExplorerProvider.rootNodeConnections;
+
+            // if a connection(s) was/were manually removed
+            if (objectExplorerConnections.length >= userConnections.length) {
+                let staleConnections = objectExplorerConnections.filter((oeConn) => {
```

Comment:
THe connections can change without the size actually changing - you should filter all the connections regardless I'd say

Reply:
What do you mean by all the connections? User connections are basically all the connections, unless I misunderstand what you're saying.

Reply:
As in just remove the length check - it's unnecessary and potentially buggy. Just do this

`const staleConnections = objectExplorerConnections.filter( oeConn => !userConnections.some(userConn => Utils.isSameConnection(oeConn, userConn)));
for(const conn of staleConnections) {
...
`

If there aren't any stale connections it just won't do anything which is fine. 

---
In reply to: [334677441](https://github.com/microsoft/vscode-mssql/pull/1423#discussion_r334677441) [](ancestors = 334677441)